ui/app/src/components/PublishDialog/PublishDialogContainer.tsx

```diff
@@ -76,9 +75,7 @@ export function PublishDialogContainer(props: PublishDialogContainerProps) {
   const [submitDisabled, setSubmitDisabled] = useState(true);
 
   const siteId = useActiveSiteId();
-  const permissionsBySite = usePermissionsBySite();
-  const myPermissions = permissionsBySite[siteId];
-  const hasPublishPermission = myPermissions?.includes('publish');
+  const hasPublishPermission = items?.every((item) => item.availableActionsMap.publish);
```

Comment:
This runs on every render; `some` would be best: doesn't iterate through the whole array and doesn't return true with an empty array
```suggestion
  const hasPublishPermission = items?.some((item) => !item.availableActionsMap.publish);
```

Reply:
It should be with a `!` at he beggining: `!items?.some((item) => !item.availableActionsMap.publish)` (if there are some items without publish permission => negate to get if it has publish permission)